src/libraries/System.Private.CoreLib/src/System/IntPtr.cs

```diff
@@ -169,10 +169,61 @@ public static int Size
         [NonVersionable]
         public unsafe void* ToPointer() => _value;
 
-        public override unsafe string ToString() =>
-            ((nint)_value).ToString(CultureInfo.InvariantCulture);
+        public static IntPtr MaxValue
+        {
+            [NonVersionable]
+            get => (IntPtr)nint.MaxValue;
+        }
+
+        public static IntPtr MinValue
+        {
+            [NonVersionable]
+            get => (IntPtr)nint.MinValue;
+        }
+
+        // Don't just delegate to nint.CompareTo as it needs to throw when not IntPtr
+        public unsafe int CompareTo(object? value)
+        {
+            if (value is null)
+            {
+                return 1;
+            }
+            if (value is IntPtr o)
+            {
+                nint i = (nint)o;
+                if ((nint)_value < i) return -1;
+                if ((nint)_value > i) return 1;
+                return 0;
+            }
+
+            throw new ArgumentException(SR.Arg_MustBeIntPtr);
+        }
+
+        public unsafe int CompareTo(IntPtr value) => ((nint)_value).CompareTo((nint)value);
+
+        [NonVersionable]
+        public unsafe bool Equals(IntPtr other) => (nint)_value == (nint)other;
+
+        public unsafe override string ToString() => ((nint)_value).ToString(CultureInfo.InvariantCulture);
```

Comment:
Test failures are because this is forcing `InvariantCulture`.

Either this needs to just mirror what `int32`/`int64` do for their respective overloads or the test needs to be updated. I would guess the former is preferred and so this should just be `((nint)_value).ToString()`

Reply:
that is a breaking change as previously it was invariant by default right?

Reply:
Yes, but it is also a breaking change to implement `IFormattable` (hence the `breaking-change` label on https://github.com/dotnet/runtime/issues/21943).

I think it is the least confusing to just make this consistent moving forward and we can direct users to explicitly specify `CultureInfo.InvariantCulture` moving forward, if they want to be backwards compatible. Otherwise, you end up with behavior that differs from the other primitive types and how these overloads generally work together in general.

(Unless @jkotas or someone else has a different opinion here).

Reply:
changed as this is already a breaking change, discussed w tanner somewhere else